specs/data_structures.md

```diff
@@ -29,45 +29,54 @@ Data Structures
-| -------------------- | ------------------------- | ----------- |
-| `version`            | [Version](#version)       |             |
-| `chainID`            | `uint64`                  |             |
-| `height`             | `uint64`                  |             |
-| `time`               | [Time](#time)             |             |
-| `lastBlockID`        | [BlockID](#blockid)       |             |
-| `lastCommitHash`     | [HashDigest](#hashdigest) |             |
-| `dataHash`           | [HashDigest](#hashdigest) |             |
-| `validatorsHash`     | [HashDigest](#hashdigest) |             |
-| `nextValidatorsHash` | [HashDigest](#hashdigest) |             |
-| `consensusHash`      | [HashDigest](#hashdigest) |             |
-| `appHash`            | [HashDigest](#hashdigest) |             |
-| `lastResultsHash`    | [HashDigest](#hashdigest) |             |
-| `evidenceHash`       | [HashDigest](#hashdigest) |             |
-| `proposerAddress`    | [Address](#address)       |             |
+Block header, which is fully downloaded by both full clients and light clients.
+
+| name                 | type                        | description                                                 |
+| -------------------- | --------------------------- | ----------------------------------------------------------- |
+| `version`            | `uint64`                    | Version of the LazyLedger chain.                            |
+| `chainID`            | `uint64`                    | Chain ID. Each fork assigns itself a (unique) ID.           |
+| `height`             | `uint64`                    | Block height. The genesis block is at height `1`.           |
+| `time`               | [Time](#time)               | Timestamp of this block.                                    |
+| `lastBlockID`        | [BlockID](#blockid)         | Previous block's ID.                                        |
+| `lastCommitRoot`     | [HashDigest](#hashdigest)   | Previous block's Tendermint commit root.                    |
+| `dataRoot`           | [HashDigest](#hashdigest)   | Message data root.                                          |
+| `validatorsRoot`     | [HashDigest](#hashdigest)   | Validator set root for this block.                          |
+| `nextValidatorsHash` | [HashDigest](#hashdigest)   | Root of the next block's validator set.                     |
+| `consensusHash`      | [HashDigest](#hashdigest)   | Consensus parameters for this block.                        |
+| `appHash`            | [HashDigest](#hashdigest)   |                                                             |
+| `lastResultsHash`    | [HashDigest](#hashdigest)   |                                                             |
+| `evidenceRoot`       | [HashDigest](#hashdigest)   | Evidence data root.                                         |
+| `codingRoots`        | [CodingRoots](#codingroots) | Commitments (i.e. Merkle roots) of the erasure-coded block. |
```

Comment:
What's the different between coding roots and data root? Aren't these the same?

Reply:
The coding roots would be Merkle roots of the erasure-coded block body, while the data root would be the Merkle root of the raw block body (well, just the message data part of it). I think the latter is needed as there's no easy way to prove inclusion of individual transactions (or messages, as it were) without the Merkle root of the raw messages.

Reply:
@musalbas says:
> The issue with this is that there is no guarantee that the raw data root
actually matches the data in the erasure coded root. Therefore there should
be only one data root.
>
> I don't see why light clients can't verify inclusion proofs in the erasure
coded root? The tree will still contain raw messages. In the LazyLedger
prototype I have light client proofs for erasure coded data roots under the
2D-RS scheme. It's more complicated but still possible.

Ah, good point, I see now.